catroid/src/main/java/org/catrobat/catroid/formulaeditor/InternFormulaKeyboardAdapter.java

```diff
@@ -456,4 +462,24 @@
 		returnList.add(new InternToken(InternTokenType.STRING, myString));
 		return returnList;
 	}
+
+	private List<InternToken> buildDoubleParameterFunction(Functions function,
+			InternTokenType firstOperator, String firstOperatorNumberValue, InternTokenType firstParameter,
```

Comment:
maybe call the `InternTokenType firstOperator` agrument `InternTokenType firstOperatorType` otherwise it's a bit harder to understand if you call it from somewhere else (same for `InternTokenType secondOperator` obiviously).

and why is `firstOperatorNumberValue` a string? you pass `Operators.MINUS.name()` for this argument, so shouldn't it be something like `operatorName`? or am I missing something? what does the `numberValue` part mean?

does it have to be `firstParameterNumberValue`? I mean, is it really true that it can only be a number? I guess not, what would you need the type for then? so if I'm not mistaken i could also be a boolean or something else, right? or a string? theoretically at least. I'd say change the name, or remove the type and hardcode the number option into the method, because then it'd be clear. either way is fine with me, just this combination is a bit misleading.

Reply:
I need to pass a minus operator and the number so that the default parameters when using it in the formulaeditor work, so my thought was to just add a possible operator to the standard `buildDoubleParameterFunction` for both numbers.
Both `buildSingleParameterFunction`and `buildDoubleParameterFunction` we pass the type and a number value so that's why I also used them but I admit the naming could have been much clearer.
If you prefer I can delete the function entirely and replace it with something like:
`public static List<InternToken> buildDoubleParameterFunction(Functions function,
			List<InternToken> firstInternTokenList, List<InternToken> secondInternTokenList) {
		List<InternToken> tokenList = new LinkedList<InternToken>();
		tokenList.add(new InternToken(InternTokenType.FUNCTION_NAME, function.name()));
		tokenList.add(new InternToken(InternTokenType.FUNCTION_PARAMETERS_BRACKET_OPEN));
		tokenList.addAll(firstInternTokenList);
		tokenList.add(new InternToken(InternTokenType.FUNCTION_PARAMETER_DELIMITER));
		tokenList.addAll(secondInternTokenList);
		tokenList.add(new InternToken(InternTokenType.FUNCTION_PARAMETERS_BRACKET_CLOSE));
		return tokenList;
	}`

Reply:
yeah you are right that does look confusing, I removed the operator type and operator string and added a boolean instead which adds a minus operator if true.